src/handshake/server.rs

```diff
@@ -28,6 +29,8 @@ use super::{
     with_first_header
 };
 
+// Most HTTP servers default to 8KB limit on headers
+const MAX_HEADERS_SIZE: usize = 8 * 1024;
```

Comment:
Should this be configurable eventually?

I guess it covers most use-cases for now

Reply:
Yeah, I didn't want to blow this PR any larger right now.